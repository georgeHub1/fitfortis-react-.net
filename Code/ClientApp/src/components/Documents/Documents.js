import React, { Component } from 'react';
import { Table, Upload, Icon, Button, List, Typography, Menu, Dropdown, Input, Modal } from 'antd';
import { FormattedMessage, injectIntl } from 'react-intl';
import { connect } from 'react-redux';
import moment from 'moment';
import styles from './styles.module.less';
import { getColorClass, sortingDocuments } from './util';
import { fetchAllDocuments, fetchRemovedDocuments, removeDocumentItem, deleteDocumentItem, restoreDocumentItem, saveDocumentSearchItem } from '../../redux/document.action';
import DocumentCreate from './DocumentCreate';
import ReactLoading from 'react-loading';
import { DocumentMobileView } from './DocumentViewMobile';
import { ReactComponent as DocumentIcon } from '../../img/documentIcon.svg';
import { ReactComponent as AudioIcon } from '../../img/audioIcon.svg';
import { ReactComponent as ImageIcon } from '../../img/imageIcon.svg';
import { ReactComponent as PdfIcon } from '../../img/pdfIcon.svg';
import { localeSet } from '../../redux/locale.action';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';

const { Search } = Input;
const Dragger = Upload.Dragger;


class Documents extends Component {
  displayName = Documents.name
   state = {
    loading: false,
    hasData: true,
    fileList: [],
    uploading: false,
    activeView: 'all',
    sorting:{
      date: 'desc',
      type: 'none',
      name: 'none'
    },
    newDocument:{
      name: '',
      extension: '',
      date: moment(),
      description: '',
      size: ''
    },
    modal:{
      isModalOpen: false,
      isEditable: false
    },
    searchText: '',
    confirmModal: false,
    selectedItem: null,
    isRestoreBtnClick: false,
    isMobile:  window.innerWidth < 768,
    newIsMobile:  window.innerWidth < 1087,
    indeterminate: false,
    checkAll: false,
    selectedType: 'date',
    focused: false
  };


  showModal = data => {
    const { modal } = this.state;

    this.setState({
      ...this.state,
      modal: {
        ...modal,
        isEditable: true,
        isModalOpen: true
      },
      newDocument: {
        ...data,
        name: data.fileName,
        extension: '',
        date: data.date,
        type: data.type.toLowerCase(),
        description: data.description,
        id: data.id,
        size: data.fileSize ? data.fileSize : ''
      }
    });
  };

  componentDidMount () {
    if (window.location.search.includes('?lang=')) {
      const lan = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

      this.props.onUpdateLanguage(lan);
      localStorage.setItem('language', lan);
    }
    this.fetchData();
    analyticId.firebaseAnalyticsLog('Documents_AllDocuments');
  }

  componentDidUpdate (prevProps) {
    const { msg } = this.props;

    if (prevProps.msg === null && msg === 'DOCUMENT_REMOVED_SUCCESS') {
      this.setState({
        ...this.state,
        confirmModal: false,
        selectedItem: null
      });
    } else if (prevProps.msg === null && msg === 'DOCUMENT_DELETE_SUCCESS') {
      this.setState({
        ...this.state,
        confirmModal: false,
        selectedItem: null
      });
    } else if (prevProps.msg === null && msg === 'SUCCESS_IN_RESTORE_DOCUMENT') {
      this.setState({
        ...this.state,
        confirmModal: false,
        selectedItem: null,
        isRestoreBtnClick: false
      });
    }
  }

  fetchData = () => {
    const { account } = this.props;

    if (account && account.id) {
      this.props.fetchAllDocuments(account.id);
      this.props.fetchRemovedDocuments(account.id);
    }
  }


  sortData = (setFilter, unsetFilter1, unsetFilter2) => {
    const { sorting } = this.state;

    if (sorting[setFilter] === 'none' || sorting[setFilter] === 'desc') {
      const obj = {...sorting, [setFilter]: 'asc', [unsetFilter1]: 'none', [unsetFilter2]: 'none' };

      this.setState({...this.state, sorting: obj});
    } else {
      const obj = {...sorting, [setFilter]: 'desc', [unsetFilter1]: 'none', [unsetFilter2]: 'none' };

      this.setState({...this.state, sorting: obj});
    }
  }

  getIcon = type => {
    const { sorting } = this.state;

    if (sorting[type] === 'none') {
      return '';
    }
    return (sorting[type] === 'asc') ?  <Icon type="up" /> : <Icon type="down" />;
  }

  /* Handle View for documents [ All / Recycle ]*/
  setActiveView = type => {
    this.setState({
      ...this.state,
      activeView: type,
      indeterminate: false,
      checkAll: false,
      selectedType: 'date'
    });
  };

  getHighlightedText = (higlight, text) => {
    const parts = text.split(new RegExp(`(${higlight.toString()})`, 'gi'));

    return (
      <span key={text}>
        {parts.map((part, i) => (
          <span
            key={i}
            // CUSTOMIZATION
            style={
              // CUSTOMIZATION: colorization of search items in Documents
              higlight !== '' && part.toLowerCase().indexOf(higlight.toLowerCase()) !== -1
                ? { fontWeight: 'bold', backgroundColor: '#FCABF9' }
                : {}
            }
          >
            {part}
          </span>
        ))}
      </span>
    );
  };

  closeModal = () => {
    this.setState({
      ...this.state,
      modal: {
        ...this.state.modal,
        isModalOpen: false
      },
      fileList: []
    });
  }

  deleteDocument = data => {
    this.setState({
      ...this.state,
      isRestoreBtnClick: false,
      confirmModal: true,
      selectedItem: data
    });
  }

  onBlur = () => {
    this.setState({ focused: false }, () => { this.onChange(); });
  }

  onFocus = () => {
    this.setState({ focused: true });
  }

  onChange = () => {
    const { activeView, sorting, searchText, focused } = this.state;
    const { documents, removedDocuments } = this.props;
    const filteredData = sortingDocuments(activeView === 'all' ? documents : removedDocuments, sorting, searchText);

    if (filteredData.length === 0 && searchText !== '' && focused) {
      this.addNewSearch(searchText, '', false);
    } else if (!focused && searchText !== '' && filteredData.length !== 0) {
      this.addNewSearch(searchText, filteredData[0].id, true);
    }
  }

  addNewSearch = (title, uniqueId, isSuccessful) => {
    const { id } = this.props.account;

    this.props.saveDocumentSearchItem(title, uniqueId, id, isSuccessful);
  };

  render () {
    const { isRestoreBtnClick, activeView, sorting, fileList, searchText, modal, confirmModal, selectedItem, isMobile, newIsMobile } = this.state;
    const { isLoading, isRestoring, isDeleting, documents, removedDocuments, isAllDocumentLoaded, isRemovedDocumentLoaded } = this.props;
    const {
      intl: { formatMessage }
    } = this.props;
    const filteredData = sortingDocuments(activeView === 'all' ? documents : removedDocuments, sorting, searchText);

    const MenuItems = ({text, data}) => (
      <Menu
        className="viewDeleteDropDown"
        subMenuCloseDelay={0.5}
        onClick={e => {
          setTimeout(() => {
            if (e.key === 'view') {
              this.showModal(data);
            } else {
              this.deleteDocument(data);
            }
          }, 100);
        }}
      >
        <Menu.Item key={'view'}>
          <span>
            <Icon type="eye" />
            {formatMessage({ id: 'Documents.view' })}
          </span>
        </Menu.Item>
        <Menu.Item key={'delete'}>
          <span >
            <Icon type="delete" />
            {formatMessage({ id: 'Documents.delete' })}
          </span>
        </Menu.Item>
      </Menu>
    );

    const RecyleMenuItems = ({text, data}) => (
      <Menu
        className="viewDeleteDropDown"
        subMenuCloseDelay={0.5}
        onClick={e => {
          setTimeout(() => {
            if (e.key === 'rollback') {
              this.setState({
                ...this.state,
                isRestoreBtnClick: true,
                confirmModal: true,
                selectedItem: data
              });
            } else if (e.key === 'delete') {
              this.deleteDocument(data);
            }
          }, 100);
        }}
      >
        <Menu.Item key={'rollback'}>
          <span>
            <Icon type="rollback" />
            {formatMessage({ id: 'Documents.rollback' })}
          </span>
        </Menu.Item>
        <Menu.Item key={'delete'}>
          <span >
            <Icon type="delete" />
            {formatMessage({ id: 'Documents.delete' })}
          </span>
        </Menu.Item>
      </Menu>
    );

    const columns = [
      {
        title: (filters, sortOrder) => {
          return (
            <span onClick={() => this.sortData('date', 'type', 'name') }>
              {formatMessage({
                id: 'Documents.date'
              })}
              {this.getIcon('date')}
            </span>
          );
        },
        dataIndex: 'date',
        key: 'date',
        width: 120,
        render: (text, record) => (
           <span className="date">{moment(new Date(text)).format('MM/DD/YYYY hh:mm a')}</span>
        )
      },
      {
        title: (filters, sortOrder) => {
          return (
            <span onClick={() => this.sortData('type', 'date', 'name') }>
              {formatMessage({
                id: 'Documents.type'
              })}
              {this.getIcon('type')}
            </span>
          );
        },
        dataIndex: 'type',
        key: 'address',
        width: 120,
        render: (text, record) => (
           <div className={`${styles.typeBox} ${styles[getColorClass(text)]}`}>
              { record.contentType.includes('image') && <Icon component={ImageIcon} /> }
              { record.contentType.includes('video') && <Icon component={AudioIcon} /> }
              { record.contentType.includes('application/pdf') && <Icon component={PdfIcon} /> }
              { !record.contentType.includes('image') && !record.contentType.includes('video') && !record.contentType.includes('application/pdf') && <Icon component={DocumentIcon} /> }
              <span className={styles.fileSize}>{record.fileSize !== '' && record.fileSize > '0' ? record.fileSize  : ''}</span>
           </div>
        )
      },
      {
        title: (filters, sortOrder) => {
          return (
            <span onClick={() => this.sortData('name', 'date', 'type') }>
              {formatMessage({
                id: 'Documents.name'
              })}
              {this.getIcon('name')}
            </span>
          );
        },
        dataIndex: 'fileName',
        key: 'name',
        render: (text, record) => (
          <span className="nameContent">
            <span onClick={() => {
              if (activeView === 'all') {
                this.showModal(record);
              }
            }}>
              <h4 className={activeView === 'all' ? '' : 'no-cursor'}>{this.getHighlightedText(searchText, text)}</h4>
            </span>
            <p>{this.getHighlightedText(searchText, record.description)}</p>
            <div className="view_delete_popup">
              <Dropdown trigger={['click']} overlay={activeView === 'all' ? <MenuItems text={text} data={record} /> : <RecyleMenuItems text={text} data={record} />} placement="bottomCenter">
                <Icon type="more" />
              </Dropdown>
            </div>
          </span>
        )
      }
    ];

    const props = {
      onRemove: file => {
        this.setState(state => {
          const index = state.fileList.indexOf(file);
          const newFileList = state.fileList.slice();

          newFileList.splice(index, 1);
          return {
            fileList: newFileList
          };
        });
      },
      beforeUpload: file => {
        const fileName = file.name;
        const ext = fileName.substr(fileName.lastIndexOf('.') + 1);
        const { modal } = this.state;

        this.setState(state => ({
          fileList: [file],
          modal:{
            ...modal,
            isEditable: false,
            isModalOpen: true
          },
          newDocument:{
            name: fileName,
            date: moment(),
            extension: ext,
            type: '',
            size: (file.size / (1024 * 1024)).toFixed(2) > 1 ? `${(file.size / (1024 * 1024)).toFixed(2)} MB` : `${(file.size / (1024)).toFixed(2)} KB`
          }
        }));
        return false;
      },
      fileList,
      multiple: false
    };

    return (
      <div>
        {
          modal.isModalOpen
          ? <DocumentCreate
            closeModal={this.closeModal}
            modal={this.state.modal}
            newDocument={this.state.newDocument}
            isLoading={isLoading}
            fileList={fileList}
          />  : null
        }
        <Modal
          title={formatMessage({ id: 'Documents.confirmationCaption' })}
          centered
          className="wrapperModal"
          visible={confirmModal}
          confirmLoading={isDeleting || isRestoring}
          okText={formatMessage({ id: `${isRestoreBtnClick ? 'Documents.rollback' : 'Documents.confirmationButtonDelete'}`})}
          cancelText={formatMessage({ id: 'Documents.confirmationButtonCancel' })}
          onOk={() => {
            if (activeView === 'all' && !isRestoreBtnClick) {
              this.props.removeDocumentItem(selectedItem.id, formatMessage);
            } else if (isRestoreBtnClick && activeView === 'recycle') {
              const { selectedItem } = this.state;
              const { account } = this.props;
              const obj = {
                  'keys': [selectedItem.id]
              };

              this.props.restoreDocumentItem(account.id, obj, formatMessage);
            } else {
              this.props.deleteDocumentItem(selectedItem.id, formatMessage);
            }
          }}
          cancelButtonProps={{
            disabled: isDeleting || isRestoring
          }}
          onCancel={() => {
            this.setState({
              ...this.state,
              confirmModal: false,
              selectedItem: null
            });
          }}
        >
          {
            (isRestoreBtnClick)
            ? <p>{formatMessage({ id: 'Documents.restoreDocument'})}</p>
            : <p>{formatMessage({ id: `${activeView === 'all' ? 'Documents.confirmationTextDeleteToRecycleBin' : 'Documents.confirmationTextFinalDelete'}` })}</p>
          }
        </Modal>
        <h1 className={styles.header}>
          <FormattedMessage id="Documents.header" />
        </h1>
        <div className={styles.mainDocuments}>
          <div className="mainUploadSection">
              <div className="uploadBox">
                <Dragger {...props} className="uploadSection">
                  <div className="uploadContent">
                    <Button>
                      <Icon type="plus" />
                      <span className="uploadText">
                        {formatMessage({ id: 'Documents.uploadText' })}
                      </span>
                    </Button>
                      {
                        !isMobile
                        ? (
                          <p>{formatMessage({ id: 'Documents.fileUploadText' })}</p>
                        ) : (
                          <p>{formatMessage({ id: 'Documents.fileUploadTextMobile' })}</p>
                        )
                      }
                  </div>
                </Dragger>
              </div>
          </div>
          <div className="documentDetails">
            <List className="documentTabs">
                <List.Item onClick={() => this.setActiveView('all')}>
                  <Typography.Text className={activeView === 'all' ? 'active' : ''}>
                    {formatMessage({ id: 'Documents.allDocuments' })} ({searchText && activeView === 'all' ? `${filteredData.length} of ${documents.length}` : documents.length})
                  </Typography.Text>
                </List.Item>
                 <List.Item onClick={() => this.setActiveView('recycle')}>
                  <Typography.Text className={activeView === 'recycle' ? 'active' : ''}>
                    {formatMessage({ id: 'Documents.recycleBin' })} ({searchText && activeView === 'recycle' ? `${filteredData.length} of ${removedDocuments.length}` : removedDocuments.length})
                  </Typography.Text>
                </List.Item>
            </List>
            <div className="searchBox">
              <Search
                placeholder={formatMessage({ id: 'Documents.searchPlaceholder' })}
                onChange={e => {
                  this.setState({
                    ...this.state,
                    searchText: e.target.value
                  }, () => { this.onChange(); });
                }}
                onBlur={this.onBlur}
                onFocus={this.onFocus}
              />
            </div>
            {
              !newIsMobile
              ? (
                <div className="userDetailsTable">
                  {
                    (isAllDocumentLoaded && isRemovedDocumentLoaded)
                      ? <Table
                          rowKey={record => record.id}
                          {...this.state}
                          columns={columns}
                          dataSource={filteredData}
                          pagination={false}
                        /> : <ReactLoading className={styles.center} type="spin" color="#335889"  />
                  }
                </div>
              ) : (
                <div>
                  {
                    (isAllDocumentLoaded && isRemovedDocumentLoaded)
                    ? <div>
                      {
                        filteredData.map((data, index) => {
                          return (
                            <DocumentMobileView
                              key={index}
                              fileName={data.fileName}
                              description={data.description}
                              contentType={data.contentType}
                              date={data.date}
                              type={data.type}
                              activeView={activeView}
                              menu={activeView === 'all' ? <MenuItems data={data} /> : <RecyleMenuItems data={data} />}
                              data={data}
                              modal={this.showModal}
                              higlightedText={this.getHighlightedText}
                              searchText={searchText}
                              selected={data.selected ? data.selected : false}
                              allData={filteredData}
                            />
                          );
                        })
                      }
                    </div>
                    : <ReactLoading className={styles.center} type="spin" color="#335889"  />
                  }
                </div>
              )
            }
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ((
  { document: { msg, isDeleting, isRestoring, removedDocuments, documents, isLoading, isAllDocumentLoaded, isRemovedDocumentLoaded }, profile : { account }  }
  ) => (
  { documents, msg, isDeleting, isRestoring, removedDocuments, isLoading, account, isAllDocumentLoaded, isRemovedDocumentLoaded }
));

const mapDispatchToProps = dispatch => ({
  fetchAllDocuments (...args) {
    return dispatch(fetchAllDocuments(...args));
  },
  fetchRemovedDocuments (...args) {
    return dispatch(fetchRemovedDocuments(...args));
  },
  removeDocumentItem (...args) {
    return dispatch(removeDocumentItem(...args));
  },
  deleteDocumentItem (...args) {
    return dispatch(deleteDocumentItem(...args));
  },
  restoreDocumentItem (...args) {
    return dispatch(restoreDocumentItem(...args));
  },
  saveDocumentSearchItem (...args) {
    return dispatch(saveDocumentSearchItem(...args));
  },
  onUpdateLanguage (args) {
    dispatch(localeSet(args));
    return Promise.resolve();
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(Documents));
