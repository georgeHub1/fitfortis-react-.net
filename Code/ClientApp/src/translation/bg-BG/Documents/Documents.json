{
  "header": "Документи",
  "uploadText": "Качи",
  "generateHealthReport": "Генерирай нов здравен отчет",
  "allDocuments": "Всички документи",
  "recycleBin": "Кошче",
  "date": "Дата",
  "type": "Вид",
  "name": "Име",
  "fileUploadText": "Придърпай документа тук или кликни, за да избереш документ от диска.",
  "fileUploadTextMobile": "Натисни тук, за да избереш документ.",
  "uploadDocument": "Качване на документ",
  "viewDocument": "Информация за документа",
  "print": "Разпечатай",
  "download": "Изтегли",
  "description": "Описание",
  "searchPlaceholder": "Търси",
  "save": "Запиши",
  "view": "Виж",
  "delete": "Изтрий",
  "cancel": "Отмени",
  "confirmationCaption": "Потвърждение",
  "confirmationTextDeleteToRecycleBin": "Наистина ли искаш да изтриеш този документ? Документът ще бъде прехвърлен в Кошчето, от където може да бъде възстановен",
  "confirmationTextFinalDelete": "Наистина ли искаш да изтриеш този документ от Кошчето? Ако го изтриеш от Кошчето, документът ще бъде загубен ЗАВИНАГИ!",
  "confirmationButtonDelete": "Изтрий",
  "confirmationButtonCancel": "Отмени",
  "selectDocumentError": "Моля избери вид на документа.",
  "nameError": "Въведи име на документа.",
  "descriptionError": "Въведи описание на документа.",
  "rollback": "Възстанови",
  "restoreDocument":"Сигурни ли сте, че искате да възстановите този документ?",
  "documentCreateSuccess":"Документът е създаден успешно.",
  "documentCreateFailed":"Създаването на документа не бе успешно.",
  "documentCreateError":"Грешка при създаването на документ.",
  "documentUpdateSuccess":"Документът е актуализиран успешно.",
  "documentUpdateFailed":"Актуализацията на документа не бе успешна.",
  "documentUpdateError":"Грешка при актуализиране на документа.",
  "documentRestoreSuccess":"Документът е възстановен успешно.",
  "documentRestoreFailed":"Възстановяването на документа не бе успешно.",
  "documentRestoreError":"Грешка при възстановяване на документа.",
  "documentDeleteSuccess":"Документът е изтрит успешно.",
  "documentDeleteFailed":"Изтриването на документа не бе успешно.",
  "documentDeleteError":"Грешка при изтриване на документа.",
  "documentDownloadSuccess": "Документът е изтеглен успешно.",
  "documentDownloadFailed": "Изтеглянето на документа не бе успешно.",
  "ok": "ОК",
  "printFailed": "Разпечатването на документа не бе успешно.",
  "printNotAllowed": "Този документ не може да се разпечата.",
  "consentAndAssentForm": "Формуляр за съгласие",
  "healthReport": "Здравен отчет",
  "insuranceReport": "Застрахователна справка",
  "medicalDiagnosis": "Мед. диагноза",
  "medicalImage": "Мед. изображение",
  "medicalTest": "Мед. тест",
  "prescription": "Рецепта",
  "other": "Друг"
}
