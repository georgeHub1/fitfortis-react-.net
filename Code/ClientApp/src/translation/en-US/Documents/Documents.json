{
  "header": "Documents",
  "uploadText": "Upload",
  "generateHealthReport": "Generate health report",
  "allDocuments": "All Documents",
  "recycleBin": "Recycle Bin",
  "date": "Date",
  "type": "Type",
  "name": "Name",
  "fileUploadText": "Drag your document here or click to upload a document from disk.",
  "fileUploadTextMobile": "Tap here to upload a document.",
  "uploadDocument": "Upload Document",
  "viewDocument": "View Document",
  "print": "Print",
  "download": "Download",
  "description": "Description",
  "searchPlaceholder": "Search",
  "save": "Save",
  "view": "View",
  "delete": "Delete",
  "cancel": "Cancel",
  "confirmationCaption": "Confirmation",
  "confirmationTextDeleteToRecycleBin": "Are you sure that you want to delete this document? The document will be moved to the Recycle Bin and you wil be able to restore it from there.",
  "confirmationTextFinalDelete": "Are you sure that you want to delete this document from the Recycle Bin? This is a final operation that CAN NOT be reverted. ",
  "confirmationButtonDelete": "Delete",
  "confirmationButtonCancel": "Cancel",
  "selectDocumentError": "Please select document type.",
  "nameError": "Please enter name.",
  "descriptionError": "Please enter description.",
  "rollback": "Restore",
  "restoreDocument":"Are you sure that you want to restore this document?",
  "documentCreateSuccess":"Document created successfully.",
  "documentCreateFailed":"Document create failed.",
  "documentCreateError":"Error while creating document.",
  "documentUpdateSuccess":"Document updated successfully.",
  "documentUpdateFailed":"Document update failed.",
  "documentUpdateError":"Error while updating document.",
  "documentRestoreSuccess":"Document restored successfully.",
  "documentRestoreFailed":"Document restore failed.",
  "documentRestoreError":"Error while restoring document.",
  "documentDeleteSuccess":"Document deleted successfully.",
  "documentDeleteFailed":"Document delete failed.",
  "documentDeleteError":"Error while delete document.",
  "documentDownloadSuccess": "Document downloaded successfully.",
  "documentDownloadFailed": "Document download failed.",
  "ok": "Ok",
  "printFailed": "Document print failed.",
  "printNotAllowed": "Can't able to print this document.",
  "consentAndAssentForm": "Consent And Assent Form",
  "healthReport": "Health Report",
  "insuranceReport": "Insurance Report",
  "medicalDiagnosis": "Medical Diagnosis",
  "medicalImage": "Medical Image",
  "medicalTest": "Medical Test",
  "prescription": "Prescription",
  "other": "Other"
}