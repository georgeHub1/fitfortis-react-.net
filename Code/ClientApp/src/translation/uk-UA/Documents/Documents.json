{
  "header": "Документи",
  "uploadText": "Завантажити",
  "generateHealthReport": "Створити звіт",
  "allDocuments": "Усі документи",
  "recycleBin": "Смітник",
  "date": "Дата",
  "type": "Тип",
  "name": "Ім'я",
  "fileUploadText": "Перетягніть тут документ або натисніть, щоб завантажити документ з диска.",
  "fileUploadTextMobile": "Клацніть тут, щоб завантажити документ.",
  "uploadDocument": "Завантажити документ",
  "viewDocument": "Переглянути документ",
  "print": "Друк",
  "download": "Завантажити",
  "description": "Опис",
  "searchPlaceholder": "Пошук",
  "save": "Зберегти",
  "view": "Вид",
  "delete": "Видалити",
  "cancel": "Скасувати",
  "confirmationCaption": "Підтвердження",
  "confirmationTextDeleteToRecycleBin":"Ви впевнені, що хочете видалити цей документ? Документ буде переміщено до смiтника, і ви зможете відновити його звідти.",
  "confirmationTextFinalDelete": "Ви впевнені, що хочете видалити цей документ із смiтника? Це остаточна операція, яку НЕ МОЖЕ бути повернено!",
  "confirmationButtonDelete": "Видалити",
  "confirmationButtonCancel": "Скасувати",
  "selectDocumentError": "Виберіть тип документа.",
  "nameError": "Будь ласка, введіть ім'я.",
  "descriptionError": "Будь ласка, введіть опис.",
  "rollback": "Відновлювати",
  "restoreDocument":"Ви впевнені, що хочете відновити цей документ?",
  "documentCreateSuccess":"Документ успішно створений.",
  "documentCreateFailed":"Не вдалося створити документ.",
  "documentCreateError":"Помилка під час створення документа.",
  "documentUpdateSuccess":"Документ успішно оновлений.",
  "documentUpdateFailed":"Помилка оновлення документа.",
  "documentUpdateError":"Помилка під час оновлення документа.",
  "documentRestoreSuccess":"Документ успішно відновлено.",
  "documentRestoreFailed":"Помилка відновлення документа.",
  "documentRestoreError":"Помилка під час відновлення документа.",
  "documentDeleteSuccess":"Документ успішно видалено.",
  "documentDeleteFailed":"Не вдалося видалити документ.",
  "documentDeleteError":"Помилка під час видалення документа.",
  "documentDownloadSuccess": "Документ успішно завантажено.",
  "documentDownloadFailed": "Помилка завантаження документа.",
  "ok": "Добре",
  "printFailed": "Не вдалося надрукувати документ.",
  "printNotAllowed": "Не вдається роздрукувати цей документ.",
  "consentAndAssentForm": "Форма згоди та згоди",
  "healthReport": "Звіт про здоров'я",
  "insuranceReport": "Звіт про страхування",
  "medicalDiagnosis": "Медична діагностика",
  "medicalImage": "Медичне зображення",
  "medicalTest": "Медичний тест",
  "prescription": "Рецепт",
  "other": "Інший"
}
