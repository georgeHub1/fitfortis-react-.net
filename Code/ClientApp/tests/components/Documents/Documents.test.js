import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter  } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Documents from '../../../src/components/Documents/Documents.js';

const Container = Documents;
const mockStore = configureStore([]);

describe('Documents', () => {
  it('renders correctly', () => {
    const account = {
      avatar: '',
      firstName: '',
      lastName: '',
      email: '',
      id: '123456'
    }
    const documents = {
      isLoading: false,
      documents: [],
      removedDocuments: [],
      msg: null,
      isAllDocumentLoaded: false,
      isRemovedDocumentLoaded: false,
      isDeleting: false
    }
    let store = mockStore({
      document: documents,
      profile: { account }
    });
    store.dispatch = jest.fn();

    const wrapper = renderer
      .create(<Provider store={store}>
          <MemoryRouter location="/documents/Medical%20service">
            <Container />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});

// 35000000  : 30 MB
// 3000000   : 2 MB

const getImage = (name, size = 3000000, mimeType) => {
  name = name || 'mock.txt';
  size = size || 500000;
  mimeType = mimeType || 'plain/txt';

  function range (count) {
    var output = '';
    for (var i = 0; i < count; i++) {
        output += 'a';
    }
    return output;
  }

  var blob = new Blob([range(size)], { type: mimeType });
  blob.lastModifiedDate = new Date();
  blob.name = name;

  return blob;
};

