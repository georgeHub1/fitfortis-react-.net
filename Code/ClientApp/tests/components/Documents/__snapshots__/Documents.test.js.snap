// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents renders correctly 1`] = `
<div>
  <h1 />
  <div>
    <div
      className="mainUploadSection"
    >
      <div
        className="uploadBox"
      >
        <span>
          <div
            className="ant-upload ant-upload-drag uploadSection"
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            style={
              Object {
                "height": undefined,
              }
            }
          >
            <span
              className="ant-upload ant-upload-btn"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              <input
                accept=""
                directory={null}
                multiple={false}
                onChange={[Function]}
                onClick={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
                type="file"
                webkitdirectory={null}
              />
              <div
                className="ant-upload-drag-container"
              >
                <div
                  className="uploadContent"
                >
                  <button
                    className="ant-btn"
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      aria-label="icon: plus"
                      className="anticon anticon-plus"
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="plus"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                        />
                        <path
                          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                        />
                      </svg>
                    </i>
                    <span
                      className="uploadText"
                    />
                  </button>
                  <p />
                </div>
              </div>
            </span>
          </div>
          <div
            className="ant-upload-list ant-upload-list-text"
          />
        </span>
      </div>
    </div>
    <div
      className="documentDetails"
    >
      <div
        className="ant-list documentTabs ant-list-split"
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
          >
            <li
              className="ant-list-item"
              onClick={[Function]}
            >
              <span
                aria-label={null}
                className="ant-typography active"
                style={
                  Object {
                    "WebkitLineClamp": null,
                  }
                }
              >
                 (
                0
                )
              </span>
            </li>
            <li
              className="ant-list-item"
              onClick={[Function]}
            >
              <span
                aria-label={null}
                className="ant-typography"
                style={
                  Object {
                    "WebkitLineClamp": null,
                  }
                }
              >
                 (
                0
                )
              </span>
            </li>
          </div>
        </div>
      </div>
      <div
        className="searchBox"
      >
        <span
          className="ant-input-search ant-input-affix-wrapper"
        >
          <input
            className="ant-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            style={null}
            type="text"
            value=""
          />
          <span
            className="ant-input-suffix"
          >
            <i
              aria-label="icon: search"
              className="anticon anticon-search ant-input-search-icon"
              onClick={[Function]}
              tabIndex={-1}
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </i>
          </span>
        </span>
      </div>
      <div>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path opacity=\\".25\\" d=\\"M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4\\"/>
  <path d=\\"M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z\\">
    <animateTransform attributeName=\\"transform\\" type=\\"rotate\\" from=\\"0 16 16\\" to=\\"360 16 16\\" dur=\\"0.8s\\" repeatCount=\\"indefinite\\" />
  </path>
</svg>
",
            }
          }
          style={
            Object {
              "fill": "#335889",
              "height": 64,
              "width": 64,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
